File: typescript/base-hmi-types/src/screens/controls/HmiRadarChartSeries.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiColor } from "../base/HmiColor.js";
import { HmiProperty } from "../base/HmiProperty.js";
import { HmiRadarChartDataPoint } from "./HmiRadarChartDataPoint.js";

export class HmiRadarChartSeries {
  index?: number;
  name?: HmiMultilingualText;
  lineAndMarker?: string;
  lineColor?: HmiProperty<HmiColor>;
  fillColor?: HmiProperty<HmiColor>;
  readonly dataPoints: HmiRadarChartDataPoint[] = [];
}

File: src/BaseHmiTypes/Screens/Controls/HmiRadarChartSeries.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiRadarChartSeries
{
    public int? Index { get; set; }

    public HmiMultilingualText? Name { get; set; }

    public string? LineAndMarker { get; set; }

    public HmiProperty<HmiColor>? LineColor { get; set; }

    public HmiProperty<HmiColor>? FillColor { get; set; }

    public IList<HmiRadarChartDataPoint> DataPoints { get; } = new List<HmiRadarChartDataPoint>();
}
